### PR TITLE
diff

### PR DESCRIPTION
**Pull request by DavidBadura**
solves #95

### CODE REVIEW

File: src/SimpleThings/AppBundle/Diff/DiffCommit.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace SimpleThings\AppBundle\Diff;
+
+use SimpleThings\AppBundle\Entity\Issue;
+
+/**
+ * @author David Badura <d.a.badura@gmail.com>
+ */
+class DiffCommit
```
Comment (tolry) [inline src/SimpleThings/AppBundle/Diff/DiffCommit.php]: Hard to think of a better name, but DiffCommit sounds a bit like an entity to me. Some brainstorming

* the class diffs commits, generates a diff from two commits, two collections of issues
* CommitDiffer?
* DiffCalculator/-Generator?
* ...

File: src/SimpleThings/AppBundle/Diff/DiffCommit.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     * @param Issue[] $a
+     * @param Issue[] $b
+     * @return Result
+     */
+    public function diff(array $a, array $b)
+    {
+        $normA = $this->norm($a);
+        $normB = $this->norm($b);
+
+        //var_dump($normA);
```
Comment (tolry) [inline src/SimpleThings/AppBundle/Diff/DiffCommit.php]: remove?

File: src/SimpleThings/AppBundle/Entity/Issue.php [hunk 3]
```diff
@@ -89,6 +93,13 @@ class Issue implements TimestampableInterface
     private $commit;
 
     /**
+     * @var string
+     *
+     * @ORM\Column(type="string")
+     */
+    private $discr;
```
Comment (tolry) [inline src/SimpleThings/AppBundle/Entity/Issue.php]: What does discr mean? discriminator, discrepency? I honestly do not know, please do not abbreviate.

File: src/SimpleThings/AppBundle/Diff/DiffCommit.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        $result->resolved = array_values($normA);
+
+        return $result;
+    }
+
+    /**
+     * @param Issue[] $issues
+     * @return Issue[]
+     */
+    private function norm(array $issues)
```
Comment (tolry) [inline src/SimpleThings/AppBundle/Diff/DiffCommit.php]: normalize

**Comment by tolry** [comment]
Looks great already :smile: What about the idea of only generating the issues differing from master in the first place? Are you planing on doing this in this PR a later PR or did you decide on abandoning the idea?

**Comment by larsborn** [comment]
that's afair still the idea: the master generates all its issues and PRs only the new and resolved ones.

File: src/SimpleThings/AppBundle/Diff/DiffCommit.php (hunk 1 above)
Comment (DavidBadura) [inline src/SimpleThings/AppBundle/Diff/DiffCommit.php]: i like DiffCalculator

**Comment by DavidBadura** [comment]
replaced by #107 and https://github.com/simpspector/analyser/pull/21

**Pull request closed without merging**